woocommerce/pinterest-for-woocommerce
# Tweak the UI of claim website

### Changes proposed in this Pull Request:

Closes #110

- Adjust the UI of website URL and error message block for the `<ClaimWebsite>`.
- Combine the processing status of claiming website to the verification button for the `<ClaimWebsite>`.
- Add JSDoc, simplify variable and replace hardcoded strings with constants for the `<ClaimWebsite>`.

### Screenshots:

#### 🎥 . Claim website success

![Kapture 2021-12-02 at 11 03 24](https://user-images.githubusercontent.com/17420811/144355557-5d4cb176-bd0b-4a12-8bce-b0ad7ff2b8be.gif)

#### 🎥 . Claim website failure

![Kapture 2021-12-02 at 12 00 09](https://user-images.githubusercontent.com/17420811/144355569-e0953930-9bfb-4544-acd1-1280cfee5616.gif)

### Detailed test instructions:

#### Prepare

If you've verified website before and want to redo the claim website process for testing, the preparing steps are:

1.  Go to https://www.pinterest.com/settings/claim and click the "Unclaim" button.
2. This plugin has cached data in the `pinterest_for_woocommerce` option of `wp_options` DB table, so we need to reset the status as well. You could update it by executing the below script on the DevTool Console:

```js
wp.apiFetch({path: '/wc-admin/options?options=pinterest_for_woocommerce'}).then((data) => {
  data.pinterest_for_woocommerce.account_data.verified_domains.length = 0;
  wp.apiFetch({
    path: '/wc-admin/options',
    method: 'POST',
    data,
  });
});
```

#### Test with a publicly accessible website

1. Go the onboarding setup: `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding&step=claim-website`
1. Before verification, the "Continue" button should be disabled.
1. Click the "Start verification" button.
1. Check if the button is displayed the correct processing status.
1. After successful verification, the "Continue" button should be enabled.
1. Refresh current page, it should show the same verified status as the previous step.
1. Click the "Continue" button and it should bring you to step 3.

#### Test with a local website

1. To test the success/failure result, it could be mocked by adding codes at line 95 in [ClaimWebsite.js](https://github.com/woocommerce/pinterest-for-woocommerce/blob/2b407c4de259d822f9b82d57b7bd20ecf9570826/assets/source/setup-guide/app/steps/ClaimWebsite.js#L95
):
    ```js
    // Simulate success result
    await new Promise( ( r ) => setTimeout( r, 1500 ) );
    setStatus( STATUS.SUCCESS );
    return;

    // Simulate failure result
    await new Promise( ( r ) => setTimeout( r, 1500 ) );
    throw {
    	data: { pinterest_code: 75 },
    	message: 'You may not have permission to this domain, or an error might have occurred. Please try again.',
    };
    ```
1. Go the onboarding setup: `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding&step=claim-website`
1. Before verification, the "Continue" button should be disabled.
1. Click the "Start verification" button.
1. Check if the button is displayed the correct processing status.
1. Check if the error message is displayed as expected.
1. After successful verification, the "Continue" button should be enabled. Click on it should bring you to step 3.

#### Check the Connection tab

The `<ClaimWebsite>` is also used in the Connection tab, and the adjusted logic associated with this page is the "Continue" button, which should not appear here.

1. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fconnection`
2. Check if "Continue" button is not shown.

### Changelog entry

> Tweak - Adjust the UI of claim website.
